sklearn/feature_extraction/_dict_vectorizer.py

```diff
@@ -78,12 +84,27 @@ class DictVectorizer(TransformerMixin, BaseEstimator):
     >>> X
     array([[2., 0., 1.],
            [0., 1., 3.]])
-    >>> v.inverse_transform(X) == \
-        [{'bar': 2.0, 'foo': 1.0}, {'baz': 1.0, 'foo': 3.0}]
+    >>> v.inverse_transform(X) == [{'bar': 2.0, 'foo': 1.0},
+    ...                            {'baz': 1.0, 'foo': 3.0}]
     True
     >>> v.transform({'foo': 4, 'unseen_feature': 3})
     array([[0., 0., 4.]])
 
+    Example with multiple values for one categorical values:
+
+    >>> D2 = [{'foo': '1', 'bar': '2'}, {'foo': '3', 'baz': '1'},
```

Comment:
I don't think normal usage would have the data sometimes contain a single value and sometimes multiple for the same key. We should demonstrate the case where 'foo' always has multiple values. For example, `{'allergies':. []}, {'allergies': ['milk', 'egg']}, {'allergies': 'milk'}`

Reply:
I'm sorry, I can't understand this comment: `allergies` is `foo` right?
So
`{foo: '1'}` -> `{allergies: 'milk'}`
`{foo: ['one', 'three']}` -> `{'allergies': ['milk', 'egg']}`
The goal is to show that it works with iterables right?

Reply:
The main point is that it would be unusual to have a dataset which, for a single input feature, like 'foo' or 'allergies' would use a string value in some data points, and a list of strings in others.
The secondary point is that 'foo' is not a nice example. I choose 'allergies' as something you could imagine in a dataset, with multiple values.

Reply:
> The main point is that it would be unusual to have a dataset which, for a single input feature, like 'foo' or 'allergies' would use a string value in some data points, and a list of strings in others.

Still not sure I'm understanding correctly, perhaps you mean that a list of strings is the most common case?

> The secondary point is that 'foo' is not a nice example. I choose 'allergies' as something you could imagine in a dataset, with multiple values.

Well, I will happily change `foo` here to `allergies` but this is true in a big number of examples in scikit-learn docstrings, and I didn't notice before similar comments... that's why I'm asking.

Anyway, thanks for your patience. I'm applying comments right now.

Reply:
> Well, I will happily change foo here to allergies but this is true in a big number of examples in scikit-learn docstrings, and I didn't notice before similar comments... that's why I'm asking.

I had previously suggested putting a bit more of an example in the narrative docs, where realistic examples are more often sought. Not needing one here...

but mostly I was trying to use the example to demonstrate to you the kind of data that this PR is useful for, and that having `{'allergy': 'milk'}` in one data point (string value) and `{'allergy': ['milk' 'eggs']}` (list value) in another data point seemed unlikely and unnecessarily confusing, but the current example with 'foo' does exactly that.

Reply:
> I had previously suggested putting a bit more of an example in the narrative docs, where realistic examples are more often sought. Not needing one here...

Ok, sorry somewhere I lost the thread probably, then...